docs/proposals/koordlet/20230728-support-cold-memory-compute.md

```diff
@@ -0,0 +1,217 @@
+
+In the Koordinator, we want to improve memory overcommitment by building the cold page reclaim ability, which can report the quantity of cold memory on the nodes and help assure that resource reclaiming like batch-memory are sound and efficient for the colocated applications.
+
+https://github.com/koordinator-sh/koordinator/issues/1178
+
+### Goals
+
+Define the API for the cold pages. Implement the cold page info collecting and reporting.
+
+### Non-Goals/Future Work
+
+Cold memory supports  scheduling optimization in koord-manager or koord-scheduler.
+
+## Proposal
+
+### Design
+
+#### Collect idle memory info and insert memory with hot page
+
+![image](../../images/support-cold-memory-1.svg)
+
+Cold memory is idle pages in page cache. This proposal applies kidled to collect idle pages. Kidled will export a file named memory.idle_stat which includes idle page. The file will exist at each hierarchical of cgroup memory. The process of collect idle memory is as shown in figure. First, kidled exports memory.idle_stat file. Second, the corresponding collector(noderesource and podresource) will collect idle info and insert metric.
+
+The proposal add a file idleinfo.go in pkg/koordlet/util/idleinfo.go
+
+Define an Idleinfo struct as follows. It corresponds to this memory.idle_stat file information.
+
+```go
+type IdleInfo struct {
+	Version             string   `json:"version"`
+	PageScans           uint64   `json:"page_scans"`
+	SlabScans           uint64   `json:"slab_scans"`
+	ScanPeriodInSeconds uint64   `json:"scan_period_in_seconds"`
+	UseHierarchy        uint64   `json:"use_hierarchy"`
+	Buckets             []uint64 `json:"buckets"`
+	Csei                []uint64 `json:"csei"`
+	Dsei                []uint64 `json:"dsei"`
+	Cfei                []uint64 `json:"cfei"`
+	Dfei                []uint64 `json:"dfei"`
+	Csui                []uint64 `json:"csui"`
+	Dsui                []uint64 `json:"dsui"`
+	Cfui                []uint64 `json:"cfui"`
+	Dfui                []uint64 `json:"dfui"`
+	Csea                []uint64 `json:"csea"`
+	Dsea                []uint64 `json:"dsea"`
+	Cfea                []uint64 `json:"cfea"`
+	Dfea                []uint64 `json:"dfea"`
+	Csua                []uint64 `json:"csua"`
+	Dsua                []uint64 `json:"dsua"`
+	Cfua                []uint64 `json:"cfua"`
+	Dfua                []uint64 `json:"dfua"`
+	Slab                []uint64 `json:"slab"`
+}
+```
+
+Define a function named readIdleInfo(). It read idleinfo memory.idle_stat and return an idleinfo object.
```

Comment:
We'd better state in which module we want to do the sys files reading and then say read which files.